doc/doxygen/scripts/set_canonical_doxygen.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2019 by the deal.II authors
+##
+## This file is part of the deal.II library.
+##
+## The deal.II library is free software; you can use it, redistribute
+## it, and/or modify it under the terms of the GNU Lesser General
+## Public License as published by the Free Software Foundation; either
+## version 2.1 of the License, or (at your option) any later version.
+## The full text of the license can be found in the file LICENSE at
+## the top level of the deal.II distribution.
+##
+## ---------------------------------------------------------------------
+
+# This script sets the canonical webpage for all the webpages of the
+# doxygen documentation. For more information see
+# https://en.wikipedia.org/wiki/Canonical_link_element
+#
+# This script is invoked by CMake. To add the canonical link to the webpages of
+# the deal.ii repository you can use the script
+# contrib/utilities/set_canonical_webpages.py
+
+import glob
+
+filenames = glob.iglob('**/*html', recursive=True)
+
+for filename in filenames:
+    file = open(filename, 'r')
```

Comment:
```suggestion
    file = open(filename, 'r', encoding='latin1')
```
Otherwise, we have problems with umlauts.

Reply:
What did you decide to do with this?